internal/engine/image_build_and_deployer.go

```diff
@@ -66,9 +66,10 @@ func (ibd *ImageBuildAndDeployer) build(ctx context.Context, service model.Servi
 		output.Get(ctx).StartPipelineStep("Building from scratch: [%s]", service.DockerfileTag)
 		defer output.Get(ctx).EndPipelineStep()
 
+		// TODO(dmiller): don't boil steps if this is an initial build
 		df := build.Dockerfile(service.DockerfileText)
-		steps := service.Steps
-		ref, err := ibd.b.BuildImageFromScratch(ctx, name, df, service.Mounts, steps, service.Entrypoint)
+		steps := model.BoilSteps(service.Steps, state.FilesChanged())
```

Comment:
I'm pretty sure that FilesChanged() is expected to be an absolute path on the local fileysystem, so i don't see how this can work?

Reply:
(it would be helpful if there was some sort of an end-to-end test, because filepath mismatch is something that we've screwed up a lot in the past)

Reply:
The PathMatchers are created in the Tiltfile with their root set to the absolute path of the cwd. Therefore the fact that FilesChanged() are absolute paths is expected and OK.

Reply:
aha! we need to be able to associate triggers both with their local paths, and with their container paths. Pausing on this while we figure out how to best support this.